atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -64,11 +64,33 @@ public ServiceDiscoveringAtlasSupplier(KeyValueServiceConfig config, Optional<Le
                         "No atlas provider for KeyValueService type " + config.type() + " could be found."
                         + " Have you annotated it with @AutoService(AtlasDbFactory.class)?"
                 ));
-        keyValueService = Suppliers.memoize(() -> atlasFactory.createRawKeyValueService(config, leaderConfig));
+        keyValueService = Suppliers.memoize(() -> pollForKeyValueService(atlasFactory));
         timestampService = () -> atlasFactory.createTimestampService(getKeyValueService());
         timestampStoreInvalidator = () -> atlasFactory.createTimestampStoreInvalidator(getKeyValueService());
     }
 
+    private KeyValueService pollForKeyValueService(AtlasDbFactory atlasFactory) {
+        long checkStartTime = System.currentTimeMillis();
+        while (true) {
+            try {
+                return atlasFactory.createRawKeyValueService(config, leaderConfig);
```

Comment:
I would prefer that we wait forever here. We dont want to kill the application as in the kubernetes world that means, that another box will come up and try to connect again. If the KVS never comes up, we are effectively shutting down and spinning new boxes (say, every 5 min).